nomad/state/state_store.go

```diff
@@ -357,6 +357,28 @@ func (s *StateStore) UpsertJob(index uint64, job *structs.Job) error {
 		return fmt.Errorf("unable to create job summary: %v", err)
 	}
 
+	// COMPAT 0.4.1 -> 0.5 Create the LocalDisk if it's nil by adding up DiskMB
+	// from task resources
+	for i, tg := range job.TaskGroups {
+		if tg.LocalDisk != nil {
+			continue
```

Comment:
Is it possible for a configuration to have both disk in Task Resources and Logical Disk, perhaps when transitioning from 0.4 -> 0.5? If so will this continue effectively ignore the 0.4 Resource config? Perhaps having both fails to validate?